workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -1,33 +1,41 @@
 using Improbable.Worker;
+using Unity.Collections;
 using Unity.Entities;
+using UnityEngine;
 
 namespace Improbable.Gdk.Core
 {
     [UpdateInGroup(typeof(SpatialOSReceiveGroup.InternalSpatialOSReceiveGroup))]
     public class SpatialOSReceiveSystem : ComponentSystem
     {
-        private WorkerBase worker;
         private Dispatcher dispatcher;
 
-        private bool inCriticalSection = false;
+        private bool inCriticalSection;
+        private string LoggerName = nameof(SpatialOSReceiveSystem);
 
-        protected override void OnCreateManager(int capacity)
+        public struct Data
         {
-            base.OnCreateManager(capacity);
+            public readonly int Length;
+            [ReadOnly] public SharedComponentDataArray<WorkerConfig> WorkerConfigs;
+        }
+
+        [Inject] private Data data;
 
-            worker = WorkerRegistry.GetWorkerForWorld(World);
+        private TranslationUnityRegistry translationUnityRegistry;
 
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            translationUnityRegistry = TranslationUnityRegistry.WorldToTranslationUnit[World];
             dispatcher = new Dispatcher();
-            SetupDispatcherHandlers();
+            SetupDispatcherHandlers();            
         }
 
+        private Worker worker;
+
         protected override void OnUpdate()
         {
-            if (worker.Connection == null)
-            {
-                return;
-            }
-
+            worker = data.WorkerConfigs[0].Worker;
```

Comment:
Even though it hasn't been our standard so far can we check that the array only contains 1 entry.

Reply:
yeah, so far I only checked the length, if necessary (because we have multiple injection groups in a system). If we only have one injection group, then that system will only run if there are any worker entities, but I can add a check for whether there is only one entry